app/services/currency_service.rb

```diff
@@ -0,0 +1,24 @@
+class CurrencyService
+  def initialize(date)
+    @date = date.blank? ? Time.now.utc.strftime('%Y-%m-%d') : date
+  end
+
+  def call
+    OpenStruct.new('success?' => true, 'errors' => nil, 'response' => JSON.parse(ApiCaller.new.call(url: build_url, http_method: :get)))
```

Comment:
As the task suggests, please first look for the record in the database first. If it is not present, please save it to the database first, then use it to find the expected rate